lucene/core/src/java/org/apache/lucene/store/MMapDirectory.java

```diff
@@ -202,24 +201,6 @@ public boolean getUseUnmap() {
     return useUnmapHack;
   }
   
-  /**
-   * Set to {@code true} to ask mapped pages to be loaded
-   * into physical memory on init. The behavior is best-effort 
-   * and operating system dependent.
-   * @see MappedByteBuffer#load
-   */
-  public void setPreload(boolean preload) {
```

Comment:
We lose the ability to always pre-load here?  But one can always make a small wrapper like `MMapDirectoryFactory` does; maybe add that as an example to `MIGRATE.txt`?

Reply:
sure I can do that

Reply:
I tend to keep the "always preload method", just deprecate it for now. I know many people that use this globally. It has the side-effect of setting madvise to "WILL_NEED" for the whole buffer (in addition to load it), so it is a bit more sticky as it's marked to the kernel as "i will need it soon".